MusicPlayer/ViewController.swift

```diff
@@ -7,23 +7,90 @@
 //
 
 import UIKit
+import AVFoundation
 
-class ViewController: UIViewController {
+class ViewController: UIViewController, AVAudioPlayerDelegate {
     
     @IBOutlet var playPauseButton: UIButton!
     @IBOutlet var timeLabel: UILabel!
     @IBOutlet var progressSliderOne: UISlider!
+    
+    // MARK: - Properties
+    var player: AVAudioPlayer!
+    var timer: Timer!
+    
+    func initializePlayer() {
+        guard let soundAsset: NSDataAsset = NSDataAsset(name: "sound") else {
+            print("음원 파일 에셋을 가져올 수 없습니다.")
+            return
+        }
+        
+        do {
+            try self.player = AVAudioPlayer(data: soundAsset.data)
+            self.player.delegate = self
+        } catch let error as NSError {
+            print("플레이어 초기화 실패")
+            print("코드 : \(error.code), 메시지 : \(error.localizedDescription)")
+        }
+        
+        self.progressSliderOne.maximumValue = Float(self.player.duration)
+        self.progressSliderOne.minimumValue = 0
+        self.progressSliderOne.value = Float(self.player.currentTime)
+    }
+    
+    func updateTimeLabelText(time: TimeInterval) {
+        let minute: Int = Int(time / 60)
+        let second: Int = Int(time.truncatingRemainder(dividingBy: 60))
+        let milisecond: Int = Int(time.truncatingRemainder(dividingBy: 1) * 100)
+        
+        let timeText: String = String(format: "%02ld:%02ld:%02ld", minute, second, milisecond)
+        
+        self.timeLabel.text = timeText
+    }
+    
+    func makeAndFireTimer() {
+        self.timer = Timer.scheduledTimer(withTimeInterval: 0.01, repeats: true, block: { [unowned self] (timer: Timer) in
+            if self.progressSliderOne.isTracking { return }
+        
+            self.updateTimeLabelText(time: self.player.currentTime)
+            self.progressSliderOne.value = Float(self.player.currentTime)
+        })
```

Comment:
호출하는 함수의 마지막 매개변수가 클로저라면 다음과 같이 표현 가능합니다!
```suggestion
        self.timer = Timer.scheduledTimer(withTimeInterval: 0.01, repeats: true) { [unowned self] (timer: Timer) in
            if self.progressSliderOne.isTracking { return }
        
            self.updateTimeLabelText(time: self.player.currentTime)
            self.progressSliderOne.value = Float(self.player.currentTime)
        }
```